interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: A parameter is information you feed into a method. While an argument is the result of a method.
 
-  Researched answer:
+  Researched answer: I turns out they are the same thing. An argument is information while a parameter mearly moving the argument from place to place
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer:.map() takes the length of the array it is .map()ing and runs what follows on all indexes of said array. I can't specify more than that without looking it up.
 
-  Researched answer:
+  Researched answer: value, index, array at least one is required while any more are only if needed. I'm unsure if you can have more than 3.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer:map runs across each and every index of an array while ending up with the same number of indexes in the array. filter uses the method to find what to keep and places that into a new array if it meets the conditions.
 
-  Researched answer:
+  Researched answer: .map iterates through an array and returns a new array of the same number of indexes. while a .filter functions through an array and returns a new array with only those values that fulfill the stated condition.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer:a method is a prebuilt function that has been added to the version of code you are working on.
 
-  Researched answer:
+  Researched answer: A function is a order of operations run when called to execute. An object is simply a function beloning to an object.
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is when you place the information across arrays instead of individual variables.
 
-  Researched answer:
+  Researched answer: Object destructuring is utilizing the information stored in an array to make quick recallable variables
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: I believe it is when you include a link between files such as with HTML and JavaScript documents.
 
-  Researched answer:
+  Researched answer: Hoisting is when you ensure information is properly ordered in the top down fashion.
```

Comment:
Hoisting in JavaScript means there is a preference that is created for functions and variable declarations.